src/Form/EntityViewDisplayEditForm.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\fb_instant_articles\Form;
+
+use Drupal\field_ui\Form\EntityViewDisplayEditForm as CoreEntityViewDisplayEditForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+
+/**
+ * Extends the core EntityViewDisplayEditForm to support multiple regions.
+ */
+class EntityViewDisplayEditForm extends CoreEntityViewDisplayEditForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTableHeader() {
+    $table_headers = parent::getTableHeader();
+    if ($this->getEntity()->getOriginalMode() === 'fb_instant_articles') {
+      $region = $this->t('Region');
+      array_splice($table_headers, 1, 0, array($region));
+    }
+    return $table_headers;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRegions() {
+    $regions = parent::getRegions();
+    if ($this->getEntity()->getOriginalMode() === 'fb_instant_articles') {
+      $new_regions['header'] = [
+        'title' => $this->t('Header'),
+        'message' => $this->t('No fields are displayed in this region.'),
+      ];
+      $new_regions['content'] = [
+        'title' => $this->t('Body'),
+        'message' => $this->t('No fields are displayed in this region.'),
+      ];
+      $new_regions['footer'] = [
+        'title' => $this->t('Footer'),
+        'message' => $this->t('No fields are displayed in this region.'),
+      ];
+      $new_regions['hidden'] = $regions['hidden'];
+      $regions = $new_regions;
+    }
+    return $regions;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function buildFieldRow(FieldDefinitionInterface $field_definition, array $form, FormStateInterface $form_state) {
```

Comment:
@bgbohling same here. I'm not following why this was added. Was something not working as expected?